pkg/utils/tekton/controller.go

```diff
@@ -96,31 +139,66 @@ func (s *SuiteController) ListTaskRuns(ns string, labelKey string, labelValue st
 	return s.PipelineClient().TektonV1beta1().TaskRuns(ns).List(context.TODO(), listOptions)
 }
 
-func (k KubeController) WatchTaskPod(tr string, taskTimeout int) error {
-	trUpdated, err := k.Tektonctrl.GetTaskRun(tr, k.Namespace)
-	if err != nil {
-		return err
-	}
-	pod, err := k.Commonctrl.GetPod(k.Namespace, trUpdated.Status.PodName)
-	if err != nil {
-		return err
-	}
-	return k.Commonctrl.WaitForPod(k.Commonctrl.IsPodSuccessful(pod.Name, k.Namespace), time.Duration(taskTimeout)*time.Second)
+func (k KubeController) WatchPipelineRun(pipelineRunName string, taskTimeout int) error {
+	return k.Commonctrl.WaitForPod(k.Tektonctrl.CheckPipelineRunFinished(pipelineRunName, k.Namespace), time.Duration(taskTimeout)*time.Second)
 }
 
-func (k KubeController) GetTaskRunResult(tr *v1beta1.TaskRun, result string) (string, error) {
-	for _, trResult := range tr.Status.TaskRunResults {
-		if trResult.Name == result {
-			return trResult.Value, nil
+func (k KubeController) GetTaskRunResult(pr *v1beta1.PipelineRun, result string) (string, error) {
+	for _, tr := range pr.Status.TaskRuns {
+		for _, trResult := range tr.Status.TaskRunResults {
+			if trResult.Name == result {
+				// for some reason the result might contain \n suffix
+				return strings.TrimSuffix(trResult.Value, "\n"), nil
+			}
 		}
 	}
 	return "", fmt.Errorf(
-		"result %q not found in TaskRun %s/%s", result, tr.ObjectMeta.Namespace, tr.ObjectMeta.Name)
+		"result %q not found in TaskRuns of PipelineRun %s/%s", result, pr.ObjectMeta.Namespace, pr.ObjectMeta.Name)
+}
+
+func (k KubeController) GetTaskRunStatus(pr *v1beta1.PipelineRun, pipelineTaskName string) (*v1beta1.PipelineRunTaskRunStatus, error) {
+	for _, tr := range pr.Status.TaskRuns {
+		if tr.PipelineTaskName == pipelineTaskName {
+			return tr, nil
+		}
+	}
+	return nil, fmt.Errorf(
+		"TaskRun status for pipeline task name %q not found in the status of PipelineRun %s/%s", pipelineTaskName, pr.ObjectMeta.Namespace, pr.ObjectMeta.Name)
 }
 
-func (k KubeController) RunTask(g TaskRunGenerator, taskTimeout int) (*v1beta1.TaskRun, error) {
-	tr := g.Generate()
-	return k.createAndWait(tr, taskTimeout)
+func (k KubeController) RunPipeline(g PipelineRunGenerator, taskTimeout int) (*v1beta1.PipelineRun, error) {
+	pr := g.Generate()
+	pvcs := k.Commonctrl.KubeInterface().CoreV1().PersistentVolumeClaims(pr.Namespace)
+	for _, w := range pr.Spec.Workspaces {
+		if w.PersistentVolumeClaim != nil {
+			pvcName := w.PersistentVolumeClaim.ClaimName
+			if _, err := pvcs.Get(context.TODO(), pvcName, metav1.GetOptions{}); err != nil {
```

Comment:
Consider extracting this into a separate helper function, e.g. `createPVCIfMissing`, for improved readability.